### PR TITLE
fix parse_links method in Resource class.

### PR DESCRIPTION
**Pull request by ArmandoMendoza**
The parse_links method in Resource class handle wrong the :href link.

### CODE REVIEW

File: lib/groovehq/resource.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def initialize(client, data)
 
     def parse_links(links)
       (links || {}).each_with_object({}) do |(relation, value), result|
-        result[relation] = Relation.new(@client, value["href"])
+        result[relation] = Relation.new(@client, value["href"] || value)
```
Comment (Fodoj) [inline lib/groovehq/resource.rb]: This change belongs to `.initialize`, where self relation is being created. This relation should initially be a regular relation with "href" attribute. Please update `initialize` to build `self` like this:

```ruby
      links            = data.delete(:links) { ActiveSupport::HashWithIndifferentAccess.new }
      links[:self]   = { href: data.delete(:href) } if data.has_key?(:href)
```

And then inside "parse_links" you can also change `value["href"]` to `value[:href]` so keys access is consistent across whole file.

Thanks! Good catch with this bug.

**Comment by Fodoj** [comment]
Please add my proposal and it's ready to be merged! I will bump gem version right after.

**Comment by ArmandoMendoza** [comment]
Sorry.. I'm waiting for this fix.

**Comment by Fodoj** [comment]
Thanks!

**Comment by Fodoj** [comment]
You can update to 1.0.3 to use fixed version :)

**Pull request merged**